visualizer.js

```diff
@@ -0,0 +1,297 @@
+  screenWidth() / 2,
+  screenHeight() / 2
+]
+
+const getcoord = (x, y) =>  [x - screenWidth() / 2, y - screenHeight() / 2]
+
+
+const getMouse = () => getcoord(mouseX, mouseY)
+
+
+function setup() {
+  createCanvas(screenWidth(), screenHeight())
+  stroke(0) // Set line drawing color to white
+  frameRate(30)
+  angleMode(DEGREES)
+  $('#container').hide()
+}
+
+const colorsTable = (data) => {
+  let colors = []
+  data.forEach((d, index) => {
+    colors.push([255 - index * 20, 0 + index * 30, 43 + index * 40])
+  })
+  return colors
+}
+
+function draw() {
+
+  background('#eee')
+  translate(...view)
+  switch (currState) {
+    case CATEGORY_MODE:
+
+      category.forEach((c, index) => {
+
+
+        push()
+        fill(...colorsTable(data)[index])
+        stroke('#333')
+        const size = c.bubble_size * (1 + (sin(rot) * 0.5)) + 20
+        ellipse(
+          ...catPos[index],
+          size,
+          size
+        )
+        pop()
+
+      })
+
+      rot = (rot + .5) % 180
+
+      push()
+      textSize(32)
+      stroke(0)
+      fill('#a463ff')
+      text(
+        'Select a Category', -(8 * 19),
+        0
+      ) // how to centering text ?
+      pop()
+      break
+
+    case ORGS_MODE:
+
+
+      let [i, j] = [0, 0]
+
+      node.forEach((n, index) => {
+        push()
+        fill(...colorsTable(data)[category.indexOf(category.find(e => e.category_name === selectedCategory.category_name))])
+        ellipse(...datPos[index], 10, 10)
+        pop()
+      })
+
+
+      break
+
+    default:
+      break
+  }
+
+
+
+  drawDesc(...trig)
+
+  $('#quit-btn').click(() => {
+
+    $('#container').slideUp()
+  })
+
+  // $('#web').click(() => {
+  //   window.location.href = 'google.com'
+  // })
+
+}
+
+
+function mouseClicked() {
+
+  let mousePos = getcoord(mouseX, mouseY)
+
+  switch (currState) {
+    case CATEGORY_MODE:
+      for (let index = 0; index < category.length; index++) {
+        if (dist(...mousePos, ...catPos[index]) <= category[index].bubble_size * (1 + (sin(rot) * 0.2))) {
+          selectedCategory = category[index]
+          currState = ORGS_MODE
+          data.forEach(e => {
+            if (e.category === selectedCategory.category_name) {
+              let tmp = [
+                random(-screenWidth() * 0.4, screenWidth() * 0.4),
+                random(-screenHeight() * 0.4, screenHeight() * 0.4)
+              ]
+
+              datPos.push(tmp)
+              node.push(e)
+            }
+          })
+          trig = [false, 0]
+          break
+        }
+      }
+      break
+    case ORGS_MODE:
+      let changeState = true
+      for (let index = 0; index < datPos.length; index++) {
+        if (dist(...mousePos, ...datPos[index]) <= 10) {
+
+          document.getElementById('title').innerHTML = node[index].name
+          document.getElementById('desc').innerHTML = node[index].tagline
+          document.getElementById('web').setAttribute('href', node[index].website_url)
```

Comment:
This still needs fixing: https://github.com/summerofcode/gsoc-orgs/pull/16#discussion_r173640150